WPF/SeeShells/SeeShells/ShellParser/OfflineRegistryReader.cs

```diff
@@ -1,15 +1,139 @@
 ﻿using System;
+using System.Collections;
 using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
 using System.Text;
-
+using Registry;
+using Registry.Abstractions;
 
 namespace SeeShells.ShellParser
 {
-    class OfflineRegistryReader : IRegistryReader
+    /// <summary>
+    /// This class is used to read an offline regisry hive file.
+    /// Using the Registry API, from EricZimmerman, it reads registry values from an hive file. 
+    /// </summary>
+    public class OfflineRegistryReader : IRegistryReader
     {
+        IConfigParser Parser { get; }
+
+        public OfflineRegistryReader(IConfigParser parser)
+        {
+            Parser = parser;
+        }
+
         public List<RegistryKeyWrapper> GetRegistryKeys()
         {
-            throw new NotImplementedException();
+            List<RegistryKeyWrapper> retList = new List<RegistryKeyWrapper>();
+
+            foreach (string location in Parser.GetLocations())
+            {
+                // Location of Offline Hive
+                var hive = new RegistryHiveOnDemand(Directory.GetParent(Directory.GetCurrentDirectory()).Parent.FullName + @"\TestResource\NTUSER.DAT");
```

Comment:
We can't have a hard reference like `@"\TestResource\NTUSER.DAT"` in the normal code thats meant for usage by the main program. 
the path to the file should be determined by something else, not OfflineRegistReader. Some other part of the code should be passing in the route for the file.
This will probably require an additional parameter in the OfflineRegistryReader's constructor that something similar to Program.cs has to pass into it.
```suggestion
                var hive = new RegistryHiveOnDemand(registryFile);
```